Feature

JstUsername opened the pull request.

JstUsername committed:
feat(.gitignore): добавлен файл .gitignore

JstUsername committed:
feat(prettier): добавлены файлы конфигурации prettier

JstUsername committed:
feat(dependencies): инициализация проекта с помощью package-lock.json, package.json

JstUsername committed:
feat(next config): добавлены файлы для конфигурации next.js

JstUsername committed:
feat(layout): добавлен компонент layout.js

JstUsername committed:
feat(theme): добавлен компонент темы theme.js

JstUsername committed:
feat(page): добавлен корневой компонент приложения page.js

JstUsername committed:
feat(navBar): добавлен компонент панели навигации navBar.js

JstUsername committed:
feat(taskList): добавлен компонент списка задач taskList.js

JstUsername committed:
feat(styles): добавлены стили для документа и скроллбара

JstUsername committed:
feat(favicon): добавлен фавикон

JstUsername committed:
fix(readme): обновлено содержимое файла README.md

JstUsername committed:
fix(readme): обновлено содержимое файла README.md

JstUsername committed:
feat(dependencies): добавлены react-hook/window-size и react-confetti

JstUsername committed:
feat(strikethrough): добавлена анимация зачеркивания текста

JstUsername committed:
feat(confetti): добавлен компонент Confetti, который активируется при удалении всех задач

IvanKoigerov commented on app/components/navBar.js:
```diff
@@ -0,0 +1,203 @@
+                onChange={handleSearch}
+                sx={{ width: '100%' }}
+              />
+            </Search>{' '}
```
`{' '}` скорее всего ide сделала, но давай уберем

IvanKoigerov commented on app/components/navBar.js:
```diff
@@ -0,0 +1,203 @@
+                inputProps={{ 'aria-label': 'search' }}
+                onChange={handleSearch}
+                sx={{ width: '100%' }}
+              />
```
У инпута есть leftIcon, поэтому лучше ее использовать

IvanKoigerov commented on app/components/navBar.js:
```diff
@@ -0,0 +1,203 @@
+import Fade from '@mui/material/Fade';
+import TextField from '@mui/material/TextField';
+
+const Search = styled('div')(({ theme }) => ({
```
Стили обычно пишут после export default, тк зачастую нужно, когда ты заходишь в компонент, то нужно посмотреть что это за компонент, а не его стили

JstUsername replied:
👍

IvanKoigerov commented on app/components/navBar.js:
```diff
@@ -0,0 +1,203 @@
+
+const Search = styled('div')(({ theme }) => ({
+  position: 'relative',
+  paddingLeft: 16,
```
Лучше везде явно использовать `theme.spacing()`

IvanKoigerov commented on app/components/navBar.js:
```diff
@@ -0,0 +1,203 @@
+  width: '100%',
+}));
+
+const SearchIconWrapper = styled('div')(({ theme }) => ({
+  height: '100%',
+  position: 'absolute',
+  pointerEvents: 'none',
+  display: 'flex',
+  alignItems: 'center',
+  justifyContent: 'center',
+}));
```
Также у нас зачастую пишут стили используя `` а не через функцию или обьект, так тебе зачастую подсказывает webstorm

JstUsername replied:
👍

IvanKoigerov commented on app/components/navBar.js:
```diff
@@ -0,0 +1,203 @@
+  },
+}));
+
+const StyledModal = {
+  position: 'absolute',
+  top: '50%',
+  left: '50%',
+  transform: 'translate(-50%, -50%)',
+  width: '50%',
+  bgcolor: 'white',
+  boxShadow: 24,
+  p: 2,
+};
```
А почему не используешь Styled?

JstUsername replied:
👍

IvanKoigerov commented on app/components/navBar.js:
```diff
@@ -0,0 +1,203 @@
+  p: 2,
+};
+
+let newTask;
+let id = 0;
```
Такой себе подход использовать таким образом переменные, у тебя тут не происходит ререндера при их изменении, не всегда это может быть нужно, также у тебя нет ни к id ни к newTask доступа кроме этого компонента.

Да и в принципе глобальные переменные создаем лишь только для более менее константных вещей, для этих вещей, лучше не создавать вещей

IvanKoigerov replied:
Используй какой-нибудь State Manager, чтобы исправить эту проблему, тот который ты ещё не использовал

JstUsername replied:
👍

IvanKoigerov commented on app/components/navBar.js:
```diff
@@ -0,0 +1,203 @@
+};
+
+export default function AppNavBar({ onClick, onChange }) {
+  const [open, setOpen] = React.useState(false);
```
При изменении этого стейта, у тебя ререндериться этот компонент, не думаю, что тебе это нужно в данном случае, подумай, как бы ты мог этого избежать

JstUsername replied:
👍

IvanKoigerov commented on app/components/navBar.js:
```diff
@@ -0,0 +1,203 @@
+
+export default function AppNavBar({ onClick, onChange }) {
+  const [open, setOpen] = React.useState(false);
+  const handleOpen = () => setOpen(true);
+  const handleClose = () => setOpen(false);
```
При ререндере этого компонента, у тебя также и заново создаются эти функции, а это ререндерит ещё раз Modal. А это тоже необходимо избежать

JstUsername replied:
👍

IvanKoigerov commented on app/components/navBar.js:
```diff
@@ -0,0 +1,203 @@
+              <StyledInputBase
+                placeholder="Search…"
+                inputProps={{ 'aria-label': 'search' }}
+                onChange={handleSearch}
```
При вводе данных, каждый раз когда ты вводишь букву, у тебя срабатывает onChange, и в этот момент у тебя ререндериться вся страница, то есть в принципе все приложение, уверен, что так и должно все работать?

IvanKoigerov commented on app/components/taskList.js:
```diff
@@ -0,0 +1,227 @@
+};
+
+export default function TaskList({ data, searchQuery }) {
+  const [complete, setComplete] = React.useState(data);
```
Советую не юзать `React.` таким образом ты в свой компонент импортируешь полностью вест пакет реакта, а нужно ли это тебе?

JstUsername replied:
👍

IvanKoigerov commented on app/components/taskList.js:
```diff
@@ -0,0 +1,227 @@
+import CloseIcon from '@mui/icons-material/Close';
+import DeleteIcon from '@mui/icons-material/DeleteOutline';
+import { useWindowSize } from '@react-hook/window-size';
+const Confetti = dynamic(() => import('react-confetti'), { ssr: false });
```
Что это такое?

JstUsername replied:
👍

IvanKoigerov commented on app/components/taskList.js:
```diff
@@ -0,0 +1,227 @@
+import { useWindowSize } from '@react-hook/window-size';
+const Confetti = dynamic(() => import('react-confetti'), { ssr: false });
+
+const StyledModal = {
+  position: 'absolute',
+  top: '50%',
+  left: '50%',
+  transform: 'translate(-50%, -50%)',
+  width: '50%',
+  bgcolor: 'white',
+  boxShadow: 24,
+  p: 2,
+};
```
Почему не юзаешь styled?

JstUsername replied:
👍

IvanKoigerov commented on app/components/taskList.js:
```diff
@@ -0,0 +1,227 @@
+                        <ListItemText
+                          id={labelId}
+                          primary={
+                            <React.Fragment>
```
Лучше вместо `React.Fragment` юазть `<></>`

JstUsername replied:
👍

IvanKoigerov commented on app/components/taskList.js:
```diff
@@ -0,0 +1,227 @@
+              .map((value) => {
+                const labelId = `checkbox-list-secondary-label-${value.id}`;
+                return (
+                  <Fade in={complete} key={value.id}>
```
Обязательно компоненты такого формата выносить в отельный компонет, иногда такого типа разделение логики тебе может сильно помочь

JstUsername replied:
👍

IvanKoigerov commented on app/globals.css:
Сам писал или в инете нашел?
Спрашиваю, потому что нужно ли это ревьювить либо нет

JstUsername replied:
👍

IvanKoigerov commented on app/layout.js:
Не уверен, лучше тут юзать лейаут или что-то другое

IvanKoigerov commented on app/page.js:
```diff
@@ -0,0 +1,25 @@
+let tasks = [];
+
+export default function Root() {
+  const [value, setValue] = useState('');
```
Не читабельно название стейта

JstUsername replied:
👍

IvanKoigerov commented on app/page.js:
```diff
@@ -0,0 +1,25 @@
+import TaskList from './components/taskList';
+import { useState } from 'react';
+
+let tasks = [];
```
Интересный способ оптимизации, но я уже писал, почему это плохо

JstUsername replied:
👍

IvanKoigerov commented on app/components/taskList.js:
```diff
@@ -0,0 +1,227 @@
+      newChecked.splice(currentIndex, 1);
+      setChecked(newChecked);
+    }
+    setComplete(data.splice(data.indexOf(value), 1));
```
В таких вещах лучше всего использовать не Data, а prev state, но у тебя тут проблема в том, как ты используешь state

У тебя есть одна общая сущность есть, это таски. А на нее ты очень много разных стейтов создал, и очень перемудренно ей манипулируешь, что в итоге приводит к нечитабельному коду и перегрузки компонентов 

Также много ререндеров

IvanKoigerov commented on app/globals.css:
```diff
@@ -0,0 +1,72 @@
+body {
+  margin: 0;
+  background-color: #f1f1f1;
```
Но цвет в body писать нельзя))

На других страницах тебе придется переписывать это значение, что было бы вообще не круто

JstUsername replied:
👍

IvanKoigerov commented on app/components/navBar.js:
Кстати, файл компонента называем также как и их нейм функции. 
`AppNavBar.js` этот файл должен называться

JstUsername replied:
👍

JstUsername committed:
fix(Component names):

названия компонентов переименованы исходя из названия функций

JstUsername committed:
feat(.idea):

инициализирован WebStorm

JstUsername committed:
fix(page.js):

исправлены импорты компонентов

JstUsername committed:
fix(global.css):

удалена установка цвета фона из body

JstUsername committed:
fix(theme.js):

исправлена палитра, добавлен оттенок dark

JstUsername committed:
fix(NavBar.js):

исправлен импорт всего пакета React, объявление стилей вынесено после компонентов, использован styled-components, цвета достаются из theme

JstUsername committed:
fix(page.js):

фавикон подключен через компонент Head, добавлен компонент Page для задания цвета фона страницы

JstUsername committed:
fix(NavBar.js):

HEX заменены на палитру theme, добавлен кастомный css для поля ввода новой задачи

JstUsername committed:
fix(TaskList):

исправлен импорт всего пакета React, стили вынесены после компонентов, добавлены кастомные стили для поля изменения задачи, элемент списка вынесен в компонент TaskItem, изменен цвет чекбоксов

JstUsername committed:
fix(StyledModal):

стили для модального окна переписаны через styled-components

JstUsername committed:
fix(CssTextField):

стилизация полей ввода переписана через styled-components

JstUsername committed:
fix(page.js):

переменная task вынесена в стейт, изменено наименование стейта, добавлены стейты и функции модального окна добавления задачи

JstUsername committed:
fix(ModalAdd):

модальное окно добавления задачи вынесено в отдельный компонент, вместо переменных использованы стейты, id задачи вычисляется по времени создания

JstUsername committed:
fix(TaskList.js):

переменная data заменена на стейт complete, скорректированы пропсы

IvanKoigerov commented:
Выложи свой пр на https://stackblitz.com, чтобы потыкаться можно было

JstUsername committed:
feat(Context.js):

добавлен компонент для описания контекстов, стейты вынесены в компонент

JstUsername committed:
feat(ModalChange.js):

добавлен отдельный компонент для модального окна изменения задачи

JstUsername committed:
fix(TaskList.js):

модальное окно изменения задачи вынесено в отдельный компонент, вместо стейтов используется контекст

JstUsername committed:
fix(ModalAdd.js):

функция onClick вынесена из корневого компонента, вместо стейтов использован контекст

JstUsername committed:
fix(NavBar.js):

вместо стейтов использован контекст

JstUsername committed:
fix(page.js):

функции разнесены по дочерним компонентам, добавлена обертка компонентов в контекст

IvanKoigerov commented on app/components/TaskList.js:
```diff
@@ -0,0 +1,103 @@
+    if (currentIndex !== -1) {
+      newChecked.splice(currentIndex, 1);
+      setChecked(newChecked);
+    }
```
Тебе нет смысла использовать несколько стейтов, ты же можешь в одном списке тудух и хранить checked или нет они

IvanKoigerov replied:
Почему так? Убираем лишний стейт, лишняя сущность, с которой надо разбираться 
Также из одного места из одного контекста ты должен мочь получить все состояние

IvanKoigerov replied:
Также это дорого в плане работ, ты делаешь 2 стейта, уже дорого, ты пишешь несколько пробегов по массиву, соответственно много циклов

JstUsername replied:
👍

IvanKoigerov commented on app/components/TaskList.js:
```diff
@@ -0,0 +1,103 @@
+    }
+  };
+
+  // const [change, setChange] = useState('');
```
коммент

IvanKoigerov commented on app/components/TaskList.js:
```diff
@@ -0,0 +1,103 @@
+      newChecked.splice(currentIndex, 1);
+      setChecked(newChecked);
+    }
+    const newComplete = complete.filter((val) => val !== value);
+    setComplete(newComplete);
```
лучше использовать `setValue(prev => )`
В prev у тебя будет всегда корректное значение, а не complete, который возможно где-то уже где-то изменился

IvanKoigerov commented on app/components/TaskList.js:
```diff
@@ -0,0 +1,103 @@
+    }
+    const newComplete = complete.filter((val) => val !== value);
+    setComplete(newComplete);
+    if (newComplete.length === 0) {
+      setRun(true);
+    }
```
Такие вещи, лучше всего выносить в useEffect

IvanKoigerov commented on app/components/TaskList.js:
```diff
@@ -0,0 +1,103 @@
+                  <Fade in={complete} key={value.id}>
+                    <TaskItem
+                      value={value}
+                      handleToggle={handleToggle}
+                      checked={checked}
+                      handleClickChange={handleClickChange}
+                      removeTask={removeTask}
```
Ты передаешь 3 функции в компонент, это значит, что эти функции у тебя будут создаваться при каждом ререндере компонента
А это не совсем круто,

В таких ситуациях мы используем `useCallback`. Чтобы избежать лишних ререндеров, да и в принципе, если куда-то передаем  функцию (в зависимости эффекта, useMemo, useCallback, в компонент и тд), то мы используем `useCallback`

IvanKoigerov commented on app/components/TaskList.js:
```diff
@@ -0,0 +1,103 @@
+
+  const [checked, setChecked] = useState([]);
+  const handleToggle = (value) => () => {
+    const currentIndex = checked.indexOf(value);
+    const newChecked = [...checked];
+    if (currentIndex === -1) {
+      newChecked.push(value);
+    } else {
+      newChecked.splice(currentIndex, 1);
+    }
+    setChecked(newChecked);
```
очень много действий делаешь 
опять же юзаем prev в этой ситуации

во вторых тебе newChecked в принципе не нужен 

я бы писал это так 
```
setChecked((prev) => {
  if (prev.some(checed => checked === value)) {
    return prev.filter(checed => checked !== value);
  }

  prev.push(value);
  return prev;
})
```

почему фильтр? потому что более читабельная вещь, тут сразу видно, что ты удаляешь элемент
почему пушу, просто дешевле, все равно запуститься ререндер компонента после вызова set, и не должно быть никаких проблем по итогу

IvanKoigerov commented on app/components/TaskList.js:
```diff
@@ -0,0 +1,103 @@
+    setChecked(newChecked);
+  };
+
+  const { width, height } = useWindowSize();
```
лучше это вынеси в отдельный компонент, тк при изменении окна у тебя будет ререндериться весь компонент туду листов, а это не круто

IvanKoigerov commented on app/page.js:
```diff
@@ -0,0 +1,42 @@
+  );
+}
+
+const Page = styled('div')`
+  margin: 0;
+  background-color: #f1f1f1;
+  height: 100vh;
+`;
```
ещё лучше бы width: 100% делать

JstUsername replied:
👍

IvanKoigerov commented on app/components/TaskItem.js:
```diff
@@ -0,0 +1,60 @@
+import Typography from '@mui/material/Typography';
+import ListItem from '@mui/material/ListItem';
+
+const TaskItem = forwardRef((props, ref) => {
+  const { value, handleToggle, checked, handleClickChange, removeTask } = props;
```
лучше сверху пропс доставать `forwardRef(({ ... }, ref)`

JstUsername replied:
👍

IvanKoigerov commented on app/components/TaskItem.js:
```diff
@@ -0,0 +1,60 @@
+import Typography from '@mui/material/Typography';
+import ListItem from '@mui/material/ListItem';
+
+const TaskItem = forwardRef((props, ref) => {
```
нет особого смысла тут юзать ref, ты же не используешь его нигде

JstUsername replied:
Ошибка, связанная с ForwardRef, поэтому в TaskItem используется forwardRef.
Ref используется в оборачивающем div.
![photo_2024-06-21_01-24-57](https://github.com/JstUsername/todo-nextjs/assets/79771449/f5b1397c-63e7-4e6b-be66-864e27e1de0f)

IvanKoigerov commented on app/components/NavBar.js:
```diff
@@ -0,0 +1,120 @@
+export default function AppNavBar({ onChange }) {
+  const setOpen = useContext(SetOpenContext);
+
+  const handleSearch = (event) => {
+    onChange(event.target.value);
+  };
```
дебаунс, а ещё лучше всего search тоже закинуть в контекст, новый соответственно

IvanKoigerov commented on app/components/ModalChange.js:
```diff
@@ -0,0 +1,131 @@
+    setChangeTask(event.target.value);
+  };
+
+  const handleSaveChange = () => {
+    if (changeTask !== change.text && changeTask !== '') {
+      setComplete(complete.map((val) => (val.id === change.id ? { id: change.id, text: changeTask } : val)));
+    }
+    setOpen(false);
+  };
+
```
тебе не нужно использовать `complete`
Как я уже писал prev

IvanKoigerov replied:
лучше всего делать ранний выход, чтобы была лучше читабельность 

ещё если текста нет, то лучше не закрывать модалку

```
if (!changeTask) return;

дальше уже пишем логику
```

Ну а по поводу того, как ты сетишь значение, я уже писал пример как это делать лучше, читабельно и красиво, попробуй повторить

IvanKoigerov commented on app/components/ModalChange.js:
По настоящему у тебя 2 одинаковых модалки) можно их обьеденить и передавать в них значение. подумай как можешь это сделать, если что пиши обсудим

IvanKoigerov commented on app/components/ModalAdd.js:
```diff
@@ -0,0 +1,132 @@
+
+  const onClick = (task) => {
+    setTask(task);
+    setComplete([...complete, task]);
```
prev, соответвенно тебе тут тоже не нужно значение всех тасок
а смысл сетить таску в этом стейте, будто бы тебе это не нужно

JstUsername replied:
👍

IvanKoigerov commented on app/components/ModalAdd.js:
```diff
@@ -0,0 +1,132 @@
+import { styled } from '@mui/material/styles';
+import { CompleteContext, OpenContext, SetCompleteContext, SetOpenContext } from './Context';
+
+let newTask;
```
переходи на стейт и сеть значение при onBlur, а не при onChange

JstUsername replied:
👍

IvanKoigerov commented on app/components/Context.js:
Все эти вещи лучше всего разделить на разные компоненты, создать папку Providers и туда все их положить

IvanKoigerov replied:
название у контекстов вообще не понятные, добавляй название сущности к которой они привязаны

ToDoContext
ModalChangeContext и тд

Старайся писать понятнее, чтобы любой другой человек мог посмотреть на переменную и понять что это и зачем она нужна, пройдись по всем

IvanKoigerov commented on app/components/Context.js:
```diff
@@ -0,0 +1,49 @@
+export const OpenContext = createContext(null);
+export const SetOpenContext = createContext(null);
+
+export const ContextAdd = ({ children }) => {
+  const [open, setOpen] = useState(false);
+  return (
+    <OpenContext.Provider value={open}>
+      <SetOpenContext.Provider value={setOpen}>{children}</SetOpenContext.Provider>
+    </OpenContext.Provider>
+  );
+};
```
Опять же нет смысла делить модалки, ну и их открытие, тоже не нужно, объединяй их

IvanKoigerov reviewed (commented):
не ставишь 👍 к коментам как я просил и ещё не скинул ссылку на StackBlitz)

JstUsername committed:
fix(dependencies):

изменены зависимости для корректной работы с stackblitz

JstUsername committed:
fix(page.js):

добавлен стиль установки ширины Page на 100%

JstUsername committed:
fix(ModalAdd.js):

вместо переменной использован стейт, две функции объединены в одну, вместо complete использован prev, событие записи новой задачи изменено на onBlur

JstUsername committed:
fix(TaskItem.js):

исправлено получение пропсов

JstUsername committed:
fix(checked):

состояние чекбоксов вынесено в complete, в функциях использован prev

JstUsername committed:
fix(TaskList.js):

передаваемые функции обернуты в useCallback

JstUsername committed:
fix(ModalAdd.js):

исправлена валидация для новой задачи

JstUsername committed:
fix(search): поиск вынесен в контекст, использован lodash debounce

JstUsername committed:
fix(Сonfetti): Confetti вынесен в отдельный компонент

JstUsername committed:
fix(Modal): модалки объединены в один компонент, использован общий стейт, переименованы контексты

JstUsername committed:
fix(Context): контексты разделены на разные компоненты, исправлены импорты

JstUsername committed:
fix(Confetti): для изменения стейта Confetti использован useEffect

JstUsername committed:
fix(import): удалены неиспользуемые хуки

JstUsername committed:
fix(providers): контекст вынесен в корень проекта, исправлены импорты

JstUsername committed:
fix(сonfetti): исправлен запуск анимации конфетти

IvanKoigerov commented on app/components/Modal.js:
```diff
@@ -0,0 +1,148 @@
+
+  // Function for Modal Add:
+  const handleAddTask = () => {
+    if (Object.keys(newTask).length !== 0 && newTask.text.length !== 0) {
```
Лучше изначальное значение делать не обьект а `undefined` или `null` 
это то чего нет 
и тогда проверять будешь `!newTask`

IvanKoigerov commented on app/components/TaskList.js:
```diff
@@ -0,0 +1,100 @@
+
+  const [run, setRun] = useState(false);
+
+  const removeTask = useCallback(
+    (value) => () => {
+      setComplete((prev) => prev.filter((val) => val !== value));
+    },
+    [complete],
+  );
+
+  const handleClickChange = useCallback(
+    (value) => () => {
+      const currentIndex = complete.indexOf(value);
+      setChange(complete[currentIndex]);
+      setOpen({ state: true, type: 'change' });
+    },
+    [complete],
+  );
```
```
useCallback((value) => {
...
}, [что-то])
```

так читабельнее

IvanKoigerov commented on app/components/Modal.js:
```diff
@@ -0,0 +1,148 @@
+  padding: 16px;
+`;
+
+const CssTextField = styled(TextField)`
+  width: 100%;
+  & label.Mui-focused {
+    color: #505050;
+  },
+& .MuiInput-underline:after {
+  border-bottom-color: #737373;
+},
+& .MuiOutlinedInput-root {
+  & fieldset {
+    border-color: #737373;
+  },
+&:hover fieldset {
+  border-color: #383838;
+},
+&.Mui-focused fieldset {
+  border-color: #505050;
+  border-width: 1px;
+},
+},
```
с отступами проблема 
делаем так
```
width: 100%;

& label.Mui-focused {
  ...
}

& .MuiOutlinedInput-root {
  & fieldset {
    ...
  }

  &:hover {
    ...
  }
}
```

IvanKoigerov commented on app/components/NavBar.js:
```diff
@@ -0,0 +1,122 @@
+  color: ${(props) => props.theme.palette.primary.dark};
+`;
+
+const StyledInputBase = styled(InputBase)(({ theme }) => ({
+  color: 'inherit',
+  '& .MuiInputBase-input': {
+    padding: theme.spacing(1, 1, 1, 0),
+    paddingLeft: 32,
+    transition: theme.transitions.create('width'),
+  },
+}));
```
а почему тут на ковычки не заменил? ну ладно, можно не фиксить

IvanKoigerov commented on app/components/TaskList.js:
```diff
@@ -0,0 +1,100 @@
+    [complete],
+  );
+
+  useEffect(() => {
+    if (complete.length === 0) {
+      setRun(true);
+    }
+  });
```
useEffect тут отрабатывает в каждом рендере, тебе это не нужно, он должен сробатывать только при изменении `complete`

IvanKoigerov commented on app/components/TaskList.js:
```diff
@@ -0,0 +1,100 @@
+        </Box>
+      </div>
+    );
+  } else {
+    return (
+      <div>
+        <Confetti run={run} />
+        <Box
+          m={3}
+          p={2}
+          display="flex"
+          flexDirection="column"
+          alignItems="center"
+          justifyContent="center"
+          borderRadius={1}
+          bgcolor={'#FFFFFF'}
+        >
+          <Typography fontSize={18} fontWeight={500} textAlign="center" color="#505050">
+            There's nothing to do today! 🎉
+          </Typography>
+          <Typography fontSize={18} fontWeight={500} textAlign="center" color="#505050">
+            Use the "New task" button to add tasks. 📝
+          </Typography>
+        </Box>
+      </div>
+    );
+  }
```
Для этой вещи лучше делать ранний выход, или же в ретер общий закинуть

IvanKoigerov commented on app/components/Modal.js:
```diff
@@ -0,0 +1,148 @@
+              <CloseIcon />
+            </IconButton>
+          </Box>
+          <Box sx={{ display: 'flex', flexWrap: 'wrap', justifyContent: 'space-between', gap: 1, mt: 1 }}>
```
Если больше 3х значение в sx, то выносим в styled

IvanKoigerov reviewed (commented):
Все остальное неплохо, есть пару ошибок, посмотри и разберись, потом будем мерджить)

JstUsername committed:
fix(useCallback): выполнен рефакторинг useCallback, удалены пустые параметры

JstUsername committed:
fix(useEffect): в зависимости добавлен complete

JstUsername committed:
fix(return): реализован ранний выход

JstUsername committed:
fix(StyledInputBase): рефакторинг стилизации с помощью styled-components

JstUsername committed:
fix(CssTextField): исправлены отступы для стилизации

JstUsername committed:
fix(FooterStyledModal): стилизация футера модального окна реализована с помощью styled-components

JstUsername committed:
fix(handleAddTask): начальное состояние стейта изменено на null, реализован ранний выход

JstUsername committed:
fix(modal title): исправлено изменение заголовка модального окна при закрытии

JstUsername committed:
fix(handleSaveChange): добавлена очистка стейта после записи измененной задачи
